### PR TITLE
added support for named parameter to the like operator of Ickle Query

### PR DESCRIPTION
**Pull request by kazuhira-r**
Using the named parameter for the like operator of the current Ickle Query throws an exception.

For example, if you run the following code
```java
   @Test
   public void testParamLike() throws Exception {
      QueryFactory qf = getQueryFactory();

      Query q = qf.create("from " + getModelFactory().getTransactionTypeName() + " where description like :descriptionParam");

      q.setParameter("descriptionParam", "%january%");

      List<Transaction> list = q.list();

      assertEquals(1, list.size());
   }
```

A ClassCastException is thrown.
```
java.lang.ClassCastException: org.infinispan.objectfilter.impl.syntax.ConstantValueExpr$ParamPlaceholder cannot be cast to java.lang.String
	at org.infinispan.objectfilter.impl.syntax.parser.QueryRendererDelegateImpl.predicateLike(QueryRendererDelegateImpl.java:308)
	at org.infinispan.objectfilter.impl.ql.parse.QueryRenderer.predicate(QueryRenderer.java:3714)
	at org.infinispan.objectfilter.impl.ql.parse.QueryRenderer.searchCondition(QueryRenderer.java:3007)
	at org.infinispan.objectfilter.impl.ql.parse.QueryRenderer.whereClause(QueryRenderer.java:657)
	at org.infinispan.objectfilter.impl.ql.parse.QueryRenderer.querySpec(QueryRenderer.java:510)
	at org.infinispan.objectfilter.impl.ql.parse.QueryRenderer.queryStatement(QueryRenderer.java:379)
	at org.infinispan.objectfilter.impl.ql.parse.QueryRenderer.queryStatementSet(QueryRenderer.java:292)
	at org.infinispan.objectfilter.impl.ql.parse.QueryRenderer.statement(QueryRenderer.java:220)
	at org.infinispan.objectfilter.impl.ql.QueryParser.render(QueryParser.java:88
```

In the QueryRendererDelegateImpl class, the value of the like operator can only handle String.
https://github.com/infinispan/infinispan/blob/9.0.0.Final/object-filter/src/main/java/org/infinispan/objectfilter/impl/syntax/parser/QueryRendererDelegateImpl.java#L307-L310

Also, I think that the reason is that the LikeExpr class can not handle the ConstantValueExpr.ParamPlaceholder class.

I think like this, but how is it?

### CODE REVIEW

File: object-filter/src/main/java/org/infinispan/objectfilter/impl/syntax/LikeExpr.java [hunk 1]
```diff
@@ -26,10 +28,27 @@ public ValueExpr getChild() {
       return child;
    }
 
-   public String getPattern() {
+   public Object getPattern() {
```
Comment (anistor) [inline object-filter/src/main/java/org/infinispan/objectfilter/impl/syntax/LikeExpr.java]: This is never used.

File: object-filter/src/main/java/org/infinispan/objectfilter/impl/syntax/LikeExpr.java [hunk 2]
```diff
@@ -26,10 +28,27 @@ public ValueExpr getChild() {
...
       return pattern;
    }
 
+   public String getPatternAs(Map<String, Object> namedParameters) {
```
Comment (anistor) [inline object-filter/src/main/java/org/infinispan/objectfilter/impl/syntax/LikeExpr.java]: I'd rename this to ```getPattern```

**Comment by anistor** [comment]
I've added a jira issue to track this : https://issues.jboss.org/browse/ISPN-7757

**Comment by anistor** [comment]
Integrated in master. Thanks @kazuhira-r !

**Comment by anistor** [comment]
I've also backported this to 9.0.x branch: https://github.com/infinispan/infinispan/pull/5101

**Comment by kazuhira-r** [comment]
Thank you for your review and integration!

**Pull request closed without merging**